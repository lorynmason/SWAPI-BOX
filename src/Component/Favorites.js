import React from 'react'
import './styles/main.scss'
import PropTypes from 'prop-types'

const Favorites = ({ favorites, toggleFavorites }) => {
  let cards;
  if(favorites.length > 0) {
      cards = favorites.map((favorite) => { 
        return(
          <div className="character-card card" key={favorite.id}>
            <button className="favorite-btn" type="button" onClick={() => toggleFavorites(favorite.id)}>
              <i className="fas fa-jedi" />
            </button>
            <h1>{favorite.name}</h1>
            <p>{favorite.homeworld}</p>
            <p>{favorite.model}</p>
            <p>{favorite.terrain}</p>
            <p>{favorite.species}</p>
            <p>{favorite.class}</p>
            <p>{favorite.passengers}</p>
            <p>{favorite.climate}</p>
            <p>{favorite.population}</p>
            <p>{favorite.residents}</p>
          </div>
        )
      })
  } else {
    cards = <div className="yoda-card card">
    <h1>selected favorites you have not</h1>
    </div>
  }
  
  return (
    <section className="cardContainer">
    {cards}
  </section>
)

}

Favorites.propTypes = {
  favorites: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired
}

export default Favorites
