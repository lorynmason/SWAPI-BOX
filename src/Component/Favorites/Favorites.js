import React from 'react'
import '../styles/main.scss'
import PropTypes from 'prop-types'

const Favorites = ({ favorites, toggleFavorites }) => {
  let cards
  if (favorites.length > 0) {
    cards = favorites.map((favorite) => {
      const infoValue = Object.values(favorite.info).map((fvInfo) => {
        return (
          <p key={favorite.info.Name}>{fvInfo}</p>
        )
      })
      return (
        <div className="character-card card" key={favorite.id}>
          <button className="favorite-btn" type="button" onClick={() => toggleFavorites(favorite.id)}>
            <i className="fas fa-jedi" />
          </button>
          <div className="info">
            <p>
              {infoValue}
            </p>
          </div>
        </div>
      )
    })
  } else {
    cards = (
      <div className="yoda-card card">
        <h1>selected favorites you have not</h1>
      </div>
    )
  }
  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Favorites.propTypes = {
  favorites: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired
}

export default Favorites
