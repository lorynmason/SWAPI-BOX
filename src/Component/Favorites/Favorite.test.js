import React from 'react'
import { shallow } from 'enzyme'
import Favorites from './Favorites'

describe('Favorites', () => {
  let wrapper
  let mockToggleFavorites
  let mockFavorites

  beforeEach(() => {
    mockFavorites = [
      {
        id: 'bob',
        category: 'characters',
        info:
        {
          Homeworld: 'Homeworld:  Earth',
          Population: 'Population:  123456789',
          Species: 'Species:  human',
          Name: 'Name:  bob'
        }
      }
    ]
    mockToggleFavorites = jest.fn(mockFavorites.id)
    wrapper = shallow(<Favorites favorites={mockFavorites} toggleFavorites={mockToggleFavorites} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleFavorites with the correct params', () => {
    wrapper.find('.favorite-btn').simulate('click')
    expect(mockToggleFavorites).toHaveBeenCalledWith(mockFavorites[0].id)
  })

  it('should render the yoda card with no favorites selected if there are no favorites AKA it shoudl match the snapshot when there are no favorites', () => {
    mockFavorites = []
    wrapper = shallow(<Favorites favorites={mockFavorites} toggleFavorites={mockToggleFavorites} />)
    expect(wrapper).toMatchSnapshot()
  })
})
