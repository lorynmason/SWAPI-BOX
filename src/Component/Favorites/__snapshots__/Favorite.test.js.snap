// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
      Array [
        Object {
          "category": "characters",
          "id": "bob",
          "info": Object {
            "Homeworld": "Homeworld:  Earth",
            "Name": "Name:  bob",
            "Population": "Population:  123456789",
            "Species": "Species:  human",
          },
        },
      ]
    }
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="character-card card"
          id="no"
        >
          <button
            className="favorite-btn"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>
          <div
            className="info"
          >
            <p>
              <p>
                Homeworld:  Earth
              </p>
              <p>
                Population:  123456789
              </p>
              <p>
                Species:  human
              </p>
              <p>
                Name:  bob
              </p>
            </p>
          </div>
        </div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "bob",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>,
            <div
              className="info"
            >
              <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>
            </div>,
          ],
          "className": "character-card card",
          "id": "no",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-jedi"
              />,
              "className": "favorite-btn",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-jedi",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>,
              "className": "info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Homeworld:  Earth
                  </p>,
                  <p>
                    Population:  123456789
                  </p>,
                  <p>
                    Species:  human
                  </p>,
                  <p>
                    Name:  bob
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Name:  bob",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Homeworld:  Earth",
                  },
                  "ref": null,
                  "rendered": "Homeworld:  Earth",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "Name:  bob",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population:  123456789",
                  },
                  "ref": null,
                  "rendered": "Population:  123456789",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "Name:  bob",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Species:  human",
                  },
                  "ref": null,
                  "rendered": "Species:  human",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "Name:  bob",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name:  bob",
                  },
                  "ref": null,
                  "rendered": "Name:  bob",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="character-card card"
            id="no"
          >
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>
            <div
              className="info"
            >
              <p>
                <p>
                  Homeworld:  Earth
                </p>
                <p>
                  Population:  123456789
                </p>
                <p>
                  Species:  human
                </p>
                <p>
                  Name:  bob
                </p>
              </p>
            </div>
          </div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "bob",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorite-btn"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-jedi"
                />
              </button>,
              <div
                className="info"
              >
                <p>
                  <p>
                    Homeworld:  Earth
                  </p>
                  <p>
                    Population:  123456789
                  </p>
                  <p>
                    Species:  human
                  </p>
                  <p>
                    Name:  bob
                  </p>
                </p>
              </div>,
            ],
            "className": "character-card card",
            "id": "no",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-jedi"
                />,
                "className": "favorite-btn",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-jedi",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    Homeworld:  Earth
                  </p>
                  <p>
                    Population:  123456789
                  </p>
                  <p>
                    Species:  human
                  </p>
                  <p>
                    Name:  bob
                  </p>
                </p>,
                "className": "info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Homeworld:  Earth
                    </p>,
                    <p>
                      Population:  123456789
                    </p>,
                    <p>
                      Species:  human
                    </p>,
                    <p>
                      Name:  bob
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "Name:  bob",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Homeworld:  Earth",
                    },
                    "ref": null,
                    "rendered": "Homeworld:  Earth",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "Name:  bob",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population:  123456789",
                    },
                    "ref": null,
                    "rendered": "Population:  123456789",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "Name:  bob",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Species:  human",
                    },
                    "ref": null,
                    "rendered": "Species:  human",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "Name:  bob",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name:  bob",
                    },
                    "ref": null,
                    "rendered": "Name:  bob",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Favorites should render the yoda card with no favorites selected if there are no favorites AKA it shoudl match the snapshot when there are no favorites 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={Array []}
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="yoda-card card"
      >
        <h1>
          selected favorites you have not
        </h1>
      </div>,
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          selected favorites you have not
        </h1>,
        "className": "yoda-card card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "selected favorites you have not",
        },
        "ref": null,
        "rendered": "selected favorites you have not",
        "type": "h1",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="yoda-card card"
        >
          <h1>
            selected favorites you have not
          </h1>
        </div>,
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            selected favorites you have not
          </h1>,
          "className": "yoda-card card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "selected favorites you have not",
          },
          "ref": null,
          "rendered": "selected favorites you have not",
          "type": "h1",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
